torch/_masked/__init__.py

```diff
@@ -686,44 +775,28 @@ def sum(input: Tensor,
         mask: Optional[Tensor] = None) -> Tensor:
     # __doc__ is generated by _apply_docstring_templates decorator
     if dtype is None:
-        dtype = input.dtype
```

Comment:
Should we be promoting int to int64 for masked_sum? Noticed that `sum` seems to do this but we have `promotes_int_to_int64=False` in the OpInfo for `_masked.sum`

Reply:
Yes, we should mirror the behavior of torch.sum. @pearu - are we missing something here?

Reply:
torch.sparse.sum handles int to int64 promotion, hence, we did not need explicit promotion here.